src/node.ts

```diff
@@ -0,0 +1,196 @@
+    'browser.display.use_document_colors': false,
+    'browser.display.use_document_fonts': 0,
+    'browser.display.use_system_colors': true,
+    'browser.formfill.enable': false,
+    'browser.helperApps.deleteTempFileOnExit': true,
+    'browser.pocket.enabled': false,
+    'browser.shell.checkDefaultBrowser': false,
+    'browser.startup.homepage': 'about:blank',
+    'browser.startup.page': 0,
+    'browser.tabs.forceHide': true,
+    'browser.urlbar.autocomplete.enabled': false,
+    'browser.urlbar.autoFill': false,
+    'browser.urlbar.showPopup': false,
+    'browser.urlbar.showSearch': false,
+    'content.notify.interval': 500000,
+    'content.notify.ontimer': true,
+    'content.switch.threshold': 250000,
+    'extensions.checkCompatibility': false,
+    'extensions.checkUpdateSecurity': false,
+    'extensions.update.autoUpdateEnabled': false,
+    'extensions.update.enabled': false,
+    'general.startup.browser': false,
+    'loop.enabled': false,
+    'network.http.pipelining.maxrequests': 8,
+    'network.http.pipelining': true,
+    'network.http.proxy.pipelining': true,
+    'permissions.default.image': 2,
+    'plugin.default_plugin_disabled': false,
+    'reader.parse-on-load.enabled': false,
+};
+
+const getPage = async () => {
+    const configFile = resolvePath('./vite.web.ts');
+    const mode = isProduction ? 'production' : 'development';
+    const useServer = !isProduction;
+
+    try {
+        const [browser, server] = await Promise.all([
+            headless.launch({
+                headless: true,
+                firefoxUserPrefs,
+            }),
+            useServer
+                ? createServer({
+                      configFile,
+                      mode,
+                      build: {
+                          target: 'modules',
+                      },
+                      define,
+                      root: rootDir,
+                  })
+                : null,
+        ]);
+
+        const [page] = await Promise.all([browser.newPage(), server?.listen()]);
+
+        server?.printUrls();
+
+        process.once('beforeExit', async () =>
+            Promise.all([browser.close(), server?.close()])
+        );
+
+        const url =
+            server?.resolvedUrls?.local[0] ??
+            pathToFileURL(resolvePath('./dist/index.html')).href;
+
+        if (url == null) {
+            throw new Error('Server startup failed');
+        }
+
+        page.on('console', (message) => {
+            console.log(message.text());
+        });
+
+        await page.goto(url);
+        await page.waitForFunction(() => typeof enketo !== 'undefined');
+
+        return page;
+    } catch (error) {
+        console.error('Launching the browser bridge failed, got error', error);
+
+        // Wait for stdout to flush.
+        await new Promise((resolve) => {
+            setTimeout(resolve, 100);
+        });
+
+        process.exit(1);
+    }
+};
+
+const pagePromise = getPage();
```

Comment:
A few things we should consider here:

- As @MartijnR mentioned, keeping a persistent browser process open runs the risk that the browser itself may crash. This seems unlikely, but we could spin up a second instance in the background which I expect would have fairly minimal impact. But it's worth measuring first to know for sure.

- I've discussed this with @lognaturel, but haven't addressed it yet: it's possible keeping a persistent _page_ open may grow in memory usage over time.  Pages are also far more crash prone. We may also want to consider periodically cycling out page instances for consistent memory usage.

- It's also not known how well a single page will handle concurrent requests. I don't believe this change meaningfully _changes_ the concurrency properties of `transform`, but it's worth considering a _pool_ of pages if that's a concern. If not, I'd similarly want to consider spinning up a second instance in the background as with the browser instance.